### ISSUE TITLE
Assign SystemBackdropConfiguration.Theme take no effect and always use SystemBackdropTheme.Dark

### ISSUE THREAD

**Issue by zhuxb711** [microsoft/WindowsAppSDK#3570]
### Describe the bug

In Windows App Sdk 1.3, a new way to use Mica or Acrylic is to set the SystemBackdrop of Window.

I created a new class to custom the Acrylic/Mica.
#### Problem 2: Change SystemBackdropConfiguration.Theme & RootLayout's RequestTheme at the same time might throw an ArgumentException. This could not be triggered every time in Demo application, please try click the button in demo multiple times with different pace.

StackTrace
```
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr)
   at ABI.Microsoft.UI.Xaml.Media.ISystemBackdropOverridesMethods.OnDefaultSystemBackdropConfigurationChanged(IObjectReference _obj, ICompositionSupportsSystemBackdrop target, XamlRoot xamlRoot)
   at Microsoft.UI.Xaml.Media.SystemBackdrop.OnDefaultSystemBackdropConfigurationChanged(ICompositionSupportsSystemBackdrop target, XamlRoot xamlRoot)
   at Microsoft.UI.Xaml.Media.SystemBackdrop.Microsoft.UI.Xaml.Media.ISystemBackdropOverrides.OnDefaultSystemBackdropConfigurationChanged(ICompositionSupportsSystemBackdrop target, XamlRoot xamlRoot)
   at ABI.Microsoft.UI.Xaml.Media.ISystemBackdropOverrides.Do_Abi_OnDefaultSystemBackdropConfigurationChanged_2(IntPtr thisPtr, IntPtr target, IntPtr xamlRoot)
```

### Steps to reproduce the bug

```csharp
namespace Test
{
    internal sealed class AcrylicMaterialBackdrop : SystemBackdrop
    {
        private readonly DesktopAcrylicController AcrylicMaterialController = new DesktopAcrylicController();
        private readonly SystemBackdropConfiguration BackdropConfiguration = new SystemBackdropConfiguration
        {
            IsInputActive = true,
            Theme = SystemBackdropTheme.Dark // Set SystemBackdropTheme.Light takes no effect
        };

        public static bool IsSupported => DesktopAcrylicController.IsSupported();

        protected override void OnTargetConnected(ICompositionSupportsSystemBackdrop connectedTarget, XamlRoot xamlRoot)
        {
            AcrylicMaterialController.AddSystemBackdropTarget(connectedTarget);
            AcrylicMaterialController.SetSystemBackdropConfiguration(BackdropConfiguration);

            base.OnTargetConnected(connectedTarget, xamlRoot);
        }

        protected override void OnTargetDisconnected(ICompositionSupportsSystemBackdrop disconnectedTarget)
        {
            AcrylicMaterialController.RemoveSystemBackdropTarget(disconnectedTarget);

            base.OnTargetDisconnected(disconnectedTarget);
        }
    }
}
```

### Expected behavior

Apply SystemBackdropTheme.Light as expected

### NuGet package version

Windows App Sdk 1.3

### Packaging type

Packaged (MSIX)

### Windows version

Windows 11 version 22H2 (22621, 2022 Update)

### IDE

Visual Studio 2022

### Additional context

Demo application: 
[BackdropTest.zip](https://github.com/microsoft/WindowsAppSDK/files/11219492/BackdropTest.zip)

**Comment by DmitriyKomin** [comment]
@zhuxb711 
_RE Problem 1: Set SystemBackdropConfiguration.Theme = SystemBackdropTheme.Light takes no effect._
This is by design . Looking at the repro app, I see that it customizes DesktopAcrylicController's properties:
```
        public AcrylicMaterialBackdrop()
        {
            AcrylicMaterialController.TintColor = Colors.SlateGray;
            AcrylicMaterialController.TintOpacity = 0.6f;
            AcrylicMaterialController.LuminosityOpacity = 0.2f;
            AcrylicMaterialController.FallbackColor = Colors.SlateGray;
        }

```
When [DesktopAcrylic/Mica]Controller properties are left at default values, the controller automatically applies Light vs Dark (hardcoded) defaults to the above material properties when the associated SystemBackdropConfiguration's Theme changes. However, if one or more properties are customized, the controller can't do this as it doesn't know what values to use for the new theme. Thus, for SystemBackdrop-based materials customizing [DesktopAcrylic/Mica]Controller properties, the material needs to manually update its properties when Theme changes. 

For standalone SystemBackdropConfiguration scenario as in the sample, this just means updating _AcrylicMaterialController.TintColor/TintOpacity/LuminosityOpacity/FallbackColor_ to match the new Theme in the AcrylicMaterialController.Theme setter (if the sample incorporated the default SystemBackdropConfiguration, it would need to override OnDefaultSystemBackdropConfigurationChanged and do the updates there).

**Comment by DmitriyKomin** [comment]
> However, docs for SystemBackdrop should mention it or the developer might be very confuse about that.

Definitely agree :-). I opened #3589 to track this doc issue.

> For Problem 2: I overrided OnDefaultSystemBackdropConfigurationChanged and do not trigger this problem any more. But still worth to investigate it.

Thanks, this is information, I agree we need to investigate. I can occasionally repro this crash locally and have dump saved, actively investigating... 

> For Problem 3 - Markup instantiation of <MicaBackdrop> 

This is a known issue; we have a fix candidate for it that we're testing internally. (For full context, in the Feb 15, 2023 WinUI Community Call <discussion #8137>, I incorrectly stated that this scenario works in light of a fix that went into WinAppSDK 1.3 Preview1 release, unfortunately that fix was not complete [fixed the crash, but material was still not getting applied] :-(.)

**Comment by zhuxb711** [comment]
Thanks @DmitriyKomin for the helpful information and transparency. I would like to close this issue now~😏
